packages/api/src/task/scheduler.ts

```diff
@@ -264,6 +277,36 @@ export class TaskScheduler {
     await this.updateTask(task, { status });
   }
 
+  async retryTask(task: WithID<Task>, event?: messages.TaskResult) {
+    let attempts = task.status.attempts + 1 ?? 1;
+    if (attempts <= MAX_RETRY_ATTEMPTS) {
+      const status: Task["status"] = {
+        ...task.status,
+        attempts: attempts,
+      };
+
+      await this.updateTask(task, { status });
+      task.status = status;
+
+      if (attempts > 1) {
+        // No timeout at first retry, 15 seconds * attempt after that
+        let retryDelay = attempts * TASK_RETRY_DELAY_COEFFICIENT;
+        await new Promise((resolve) => setTimeout(resolve, retryDelay));
```

Comment:
https://github.com/livepeer/livepeer-com/blob/88236319556da6cf0c1c4f0dc9fa3b89d9c87a0c/packages/api/src/util.ts#L24
```suggestion
        await sleep(retryDelay);
```

Reply:
Btw I actually loved the "sleep retry", I was biased having implemented the healthcheck logic on the task-runner through delayed AMQP events, and was thinking we'd need that complexity here as well. For such short delays it seems pretty much fine to just sleep on the event and retry tho. There might be a scalability problem if we have too many of these tasks retrying tho, each retry is effectively blocking 1 API consumer from processing other events, but I believe it works great for now.

Reply:
Yeah, the task runner should receive the failing back and fire the retry right?

Reply:
The task runner fires the failing, the API fires the retry, the task-runner will pick up the retry as just another task execution.